test/cql-pytest/test_materialized_view.py

```diff
@@ -992,3 +992,209 @@ def test_base_clustering_prefix_deletion(cql, test_keyspace):
+
+            # After the deletion, all data should be gone from both base and view
+            assert [] == list(cql.execute(f"SELECT v FROM {table} WHERE p=1 AND c=10"))
+            assert [] == list(cql.execute(f"SELECT v FROM {mv} WHERE c=10 AND p=1"))
+            assert allN == [x.v for x in cql.execute(f"SELECT v FROM {table} WHERE p=2 AND c=20")]
+            assert allN == sorted([x.v for x in cql.execute(f"SELECT v FROM {mv} WHERE c=20 AND p=2")])
+
+            metrics_after = ScyllaMetrics.query(cql)
+            updates_after = metrics_after.get('scylla_database_total_view_updates_pushed_local')
+
+            print(f"scylla_database_total_view_updates_pushed_local: {updates_before} -> {updates_after}")
+            assert updates_after == updates_before + 1
+
+# Perform a batch operation, deleting a partition and also inserting a row
+# to that partition with a newer timestamp, and verify that the insertion
+# is not lost in the MV update.
+def test_base_partition_deletion_in_batch_with_insert(cql, test_keyspace):
+    with new_test_table(cql, test_keyspace, 'p int, c int, primary key (p,c)') as table:
+        insert = cql.prepare(f'INSERT INTO {table} (p,c) VALUES (?,?) USING TIMESTAMP 99')
+        with new_materialized_view(cql, table, '*', 'p,c', 'p is not null and c is not null') as mv:
+            cql.execute(insert, [0, 1])
+            cql.execute(insert, [0, 2])
+            cql.execute(insert, [0, 3])
+
+            # This should delete all the existing partition rows, and the new
+            # row insertion survives and remains the only row after the operation
+            # since it has the most recent timestamp.
+            cmd = 'BEGIN UNLOGGED BATCH '
+            cmd += f'DELETE FROM {table} USING TIMESTAMP 100 WHERE p=0; '
+            cmd += f'INSERT INTO {table} (p,c) VALUES (0,4) USING TIMESTAMP 101; '
+            cmd += 'APPLY BATCH;'
+            cql.execute(cmd)
+
+            # Verify it is correct both in the table and the view
+            assert [4] == [x.c for x in cql.execute(f"SELECT c FROM {table} WHERE p=0")]
+            assert [4] == [x.c for x in cql.execute(f"SELECT c FROM {mv} WHERE p=0")]
+
+# Similar to the test above, perform a deletion of a base partition in a batch with
+# deletion of individual rows. Verify the partition is deleted correctly and that
+# a single update is generated for the view for deleting the whole partition, and no
+# view updates for each row.
+def test_base_partition_deletion_in_batch_with_delete_row_metrics(cql, test_keyspace, scylla_only):
+    with new_test_table(cql, test_keyspace, 'p int, c int, v int, primary key ((p,c),v)') as table:
+        insert = cql.prepare(f'INSERT INTO {table} (p,c,v) VALUES (?,?,?)')
+        # The view partition key is the same as the base partition key.
+        with new_materialized_view(cql, table, '*', '(p,c),v', 'p is not null and c is not null and v is not null') as mv:
+            N = 1000
```

Comment:
Again, please reconsider the high number. No need for the test to take 0.5 seconds if it can take 0.1 seconds.